test/mysql.test.js

```diff
@@ -425,6 +436,77 @@ describe('mysql', function() {
     });
   });
 
+  it('should handle null in inq operator', function(done) {
+    PostWithNumId.create({id: 1, title: 'Foo', content: 'Bar'}, function(err, post) {
+      should.not.exist(err);
+      PostWithNumId.find({where: {id: {inq: [null, 1]}}}, function(err, posts) {
+        should.not.exist(err);
+        posts.length.should.equal(1);
+        posts[0].title.should.equal('Foo');
+        posts[0].id.should.equal(1);
+        done();
+      });
+    });
+  });
+
+  it('should handle null in nin operator', function(done) {
+    PostWithNumId.create({id: 2, title: 'Make', content: 'Toyota'}, function(err, post) {
+      should.not.exist(err);
+      PostWithNumId.find({where: {id: {nin: [null, 3]}}}, function(err, posts) {
+        should.not.exist(err);
+        posts.length.should.equal(1);
+        posts[0].content.should.equal('Toyota');
+        posts[0].id.should.equal(2);
+        done();
+      });
+    });
+  });
+
+  it('should handle null in neq operator', function(done) {
+    PostWithNumId.create({id: 3, title: 'Model', content: 'Corolla'}, function(err, post) {
+      should.not.exist(err);
+      PostWithNumId.find({where: {id: {neq: null}}}, function(err, posts) {
+        should.not.exist(err);
+        posts.length.should.equal(1);
+        posts[0].content.should.equal('Corolla');
+        posts[0].id.should.equal(3);
+        done();
+      });
+    });
+  });
+
+  it('should handle null in inq op for different datatypes', function(done) {
+    var defaultPost = {
+      id: 0,
+      title: 'defTitle',
+      content: 'defContent',
+      buffProp: new Buffer('defBuffer'),
+      objProp: {defKey: 'defVal'},
+      arrProp: [0],
+      dateProp: new Date(),
+      pointProp: {lng: 4.51515, lat: 57.2314},
+    };
+
+    PostWithNumId.create(defaultPost, function(err, post) {
+      should.not.exist(err);
+      PostWithNumId.find({where: {and: [
+        {id: {inq: [null]}},
+        {title: {inq: [null]}},
+        {content: {inq: [null]}},
+        {buffProp: {inq: [null]}},
+        {objProp: {inq: [null]}},
+        {arrProp: {inq: [null]}},
+        {dateProp: {inq: [null]}},
+        {pointProp: {inq: [null]}},
+      ]}}, function(err, posts) {
+        //the main check here is that we produce a valid sql
```

Comment:
I don't think this is true.
I understand that syntactically invalid SQL would throw an error, but you could easily have a logic error that produced _semantically_ invalid SQL that does something, just not the _right_ thing.